lib/sles4sap_publiccloud.pm

```diff
@@ -997,4 +998,41 @@ sub saphanasr_showAttr_version {
     }
 }
 
+=head2 wait_for_zypper
+
+    The function attempts to run 'zypper lr' to check for a lock. If Zypper is locked, it waits for a specified delay before retrying.
+    Returns normally if Zypper is not locked, or dies after a maximum number of retries if Zypper remains locked.
```

Comment:
here and there `zypper` is named with capital first letter `Zypper`. Is it ok to do so?